src/utils.jl

```diff
@@ -21,6 +21,22 @@ function unique_indices(x)
     return unique, indices
 end
 
+"""
+    split_chains(data::AbstractArray{<:Any,3}, split::Int=2)
+
+Split each chain in `data` of shape `(ndraws, nchains, nparams)` into `split` chains.
+
+If `ndraws` is not divisible by `split`, the last `mod(ndraws, split)` iterations are
+dropped. The result is a reshaped view of `data`.
+"""
+function split_chains(data::AbstractArray{<:Any,3}, split::Int=2)
+    ndraws, nchains, nparams = size(data)
+    ndraws_split, niter_drop = divrem(ndraws, split)
+    nchains_split = nchains * split
+    data_sub = @views data[firstindex(data):(end - niter_drop), :, :]
```

Comment:
Shouldn't this be the first index along the first axis only?
```suggestion
    data_sub = @views data[firstindex(data, 1):(end - niter_drop), :, :]
```
 Or on newer Julia versions simply
```suggestion
    data_sub = @views data[begin:(end - niter_drop), :, :]
```

Reply:
`begin` isn't supported on Julia v1.3 (our oldest supported version). I didn't even know `firstindex` could take a dim argument.